src/EditorFeatures/Core/EditorConfigSettings/Data/CodeStyle/CodeStyleSetting.PerLanguageEnumCodeStyleSetting.cs

```diff
@@ -60,6 +65,38 @@ protected override CodeStyleOption2<T> GetOption()
                     // TODO(jmarolf): Should we expose duplicate options if the user has a different setting in VB vs. C#?
                     //                Today this code will choose whatever option is set for C# as the default.
                     : _visualStudioOptions.GetOption<CodeStyleOption2<T>>(new OptionKey2(_option, LanguageNames.CSharp));
+
+            public override string? GetSettingName()
+            {
+                var storageLocation = GetEditorConfigStorageLocation(_option);
+                return storageLocation?.KeyName;
+            }
+
+            public override string GetDocumentation()
+            {
+                return Description;
+            }
+
+            public override ImmutableArray<string>? GetSettingValues(OptionSet optionSet)
+            {
+                var type = typeof(T);
+                var strings = new List<string>();
+                var enumValues = type.GetEnumValues();
+
+                foreach (var enumValue in enumValues)
```

Comment:
can we share most of the code with the one in [EnumCodeStyleSetting.cs](https://github.com/dotnet/roslyn/pull/63019/files#diff-6fb21c93a92a3fcf3ce0f3d4cc180e630e9ee595b46d80c4901504591174541a)